conans/test/functional/generators/cmake_find_package_multi_test.py

```diff
@@ -526,3 +526,121 @@ def build(self):
         self.assertIn("component libs: $<$<CONFIG:Debug>:;>;$<$<CONFIG:MinSizeRel>:;>;"
                       "$<$<CONFIG:RelWithDebInfo>:;>;$<$<CONFIG:Release>:system_lib_component;",
                       t.out)
+
+
+@pytest.mark.tool_cmake
+class TestNoNamespaceTarget:
+    """ This test case uses build-modules feature to create a target without a namespace. This
+        target uses targets create by Conan (build_modules are included after Conan targets)
+    """
+
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile, CMake
+
+        class Recipe(ConanFile):
+            settings = "os", "compiler", "arch"
+            exports_sources = ["src/*", "build-module.cmake"]
+            generators = "cmake"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure(source_folder="src")
+                cmake.build()
+
+            def package(self):
+                self.copy("*.h", dst="include", src="src")
+                self.copy("*.lib", dst="lib", keep_path=False)
+                self.copy("*.dll", dst="bin", keep_path=False)
+                self.copy("*.dylib*", dst="lib", keep_path=False)
+                self.copy("*.so", dst="lib", keep_path=False)
+                self.copy("*.a", dst="lib", keep_path=False)
+                self.copy("build-module.cmake", dst="share/cmake")
+
+            def package_info(self):
+                self.cpp_info.libs = ["library"]
+                module = os.path.join("share", "cmake", "build-module.cmake")
+                self.cpp_info.build_modules = module
+    """)
+
+    build_module = textwrap.dedent("""
+        message(">> Build-module is included")
+
+        if(NOT TARGET nonamespace)
+            add_library(nonamespace INTERFACE IMPORTED)
+            target_link_libraries(nonamespace INTERFACE library::library)
+        endif()
+    """)
+
+    consumer = textwrap.dedent("""
+        cmake_minimum_required(VERSION 3.0)
+        set(CMAKE_CXX_COMPILER_WORKS 1)
+        set(CMAKE_CXX_ABI_COMPILED 1)
+        project(consumer)
+
+        find_package(library)
+
+        get_target_property(LIBS1 library::library INTERFACE_LINK_LIBRARIES)
+        message(">> library::library libs: ${LIBS1}")
+
+        get_target_property(LIBS2 nonamespace INTERFACE_LINK_LIBRARIES)
+        message(">> nonamespace libs: ${LIBS2}")
+
+        add_executable(consumer main.cpp)
+        target_link_libraries(consumer nonamespace)
+    """)
+
+    main = textwrap.dedent("""
+        #include "library.h"
+
+        int main() {
+            library();
+        }
+    """)
+
+    @classmethod
```

Comment:
why not using pytest fixtures? I have started to used them and so far look great, if we are going pytest, probably we want to start adopting its practices?

Reply:
because the purpose here is to execute these lines once for all the tests in this class.

A fixture typically provides a way to execute (and inject) something before/after tests, but IMHO **something intended to be reusable by many tests in the test suite**. This one probably applies only to this test-class, I think these class methods are better than fixtures when they apply only to one specific test.